Sources/FoundationExtensions/Decoder+Extensions.swift

```diff
@@ -123,7 +123,11 @@ extension JSONEncoder {
 
 extension ErrorUtils {
 
-    static func logDecodingError(_ error: Error, type: Any.Type) {
+    static func logDecodingError(_ error: Error, type: Any.Type, data: Data? = nil) {
+        if let data = data {
+            Logger.debug(Strings.codable.invalid_data_when_decoding(data))
```

Comment:
Hmm so this will be a separate log. I wonder if it should be part of the error log below...

Reply:
I did it this way for 2 reasons, let me know what you think:

- The implementation of this method uses like 5 different strings so I'd have to modify all of them
- The JSON might contain sensitive data, so we might not want the whole thing to be in the console. In release builds, `debug` logs aren't enabled by default, therefore eliminating this.

Reply:
Ah ok makes sense 👍